angular/projects/spark-angular/src/lib/components/sprk-autocomplete/sprk-autocomplete-result/sprk-autocomplete-result.directive.ts

```diff
@@ -0,0 +1,154 @@
+import {
+  Directive,
+  ElementRef,
+  OnInit,
+  HostListener,
+  Output,
+  EventEmitter,
+  Input,
+  HostBinding,
+  Renderer2,
+} from '@angular/core';
+import { uniqueId } from 'lodash';
+
+@Directive({
+  selector: '[sprkAutocompleteResult]',
+})
+export class SprkAutocompleteResultDirective implements OnInit {
+  /**
+   * @ignore
+   */
+  constructor(public ref: ElementRef, private renderer: Renderer2) {}
+
+  /**
+   * The value supplied will be assigned
+   * to the `data-id` attribute on the
+   * element. This is intended to be
+   * used as a selector for automated
+   * tools. This value should be unique
+   * per page.
+   */
+  @HostBinding('attr.data-id')
+  @Input()
+  idString: string;
+
+  /**
+   * The value supplied will be assigned to the
+   * `data-analytics` attribute on the element.
+   * Intended for an outside
+   * library to capture data.
+   */
+  @HostBinding('attr.data-analytics')
+  @Input()
+  analyticsString: string;
+
+  /**
+   * Expects a space separated string
+   * of classes to be added to the
+   * element.
+   */
+  @Input()
+  additionalClasses: string;
+
+  /**
+   * @ignore
+   *
+   */
+  private _isHighlighted: boolean;
+
+  /**
+   * If true, this Autocomplete result will be highlighted.
+   */
+  @HostBinding('class.sprk-c-Autocomplete__result--active')
```

Comment:
This is the only `@HostBinding('class` in this PR. The other classes are added using `getClasses` functions.

Reply:
I see, since `aria-selected` and `--active` class are both changed upon, `isHighlighted`, you combined them.

I think that makes sense because they keep things closer together.
One thing I notice about `HostBinding` is that it adds the class to the custom element, not the actual HTML Element. But since this is a directive, this should be fine because it directly adds it to the element it's on.

https://codesandbox.io/s/elastic-river-7jw6f?file=/src/app/app.component.ts
When you click "activate" you see it only adds it to the custom element...which is not great.

Reply:
Yes I think you're correct. Putting the class on the custom element is not a good idea because those custom elements are not valid HTML elements.